### ISSUE TITLE
[core] IsKeyDown() not working properly

### ISSUE THREAD

**Issue by raysan5** [raysan5/raylib#379]
It seems `IsKeyDown()` is reseted for one frame, not returning the correct value during intermitent frames.

In Raspberry Pi, keyboard input is `read()` in raw mode and data processed in function [`ProcessKeyboard()`](https://github.com/raysan5/raylib/blob/master/src/core.c#L2982), that function should be reviewed because something is not working properly...

**Comment by raysan5** [comment]
It could be related to some points commented on issue https://github.com/raysan5/raylib/issues/679

**Comment by DarkElvenAngel** [comment]
@raysan5 
I've been looking into this as RPI is my target platform.
I have found problems and possibly a solution, I just have no idea how to implement it.
The problem is STDIN processes key repeats and you don't get true raw reads from the keyboard.  I've been looking into how to read the events directly from the /dev/input/eventX these can be filtered to remove key repeats.  I have been reading through core.c to find where I and if this would work.
Raylib can already detect the correct eventX linked to my keyboard and I have code to read from this event.  I just don't see how to get this into the library correctly.
```c
#include <stdio.h>
#include <fcntl.h>
#include <linux/input.h>
#include <unistd.h>
#include <signal.h>
#include <stdlib.h>

void INThandler(){
        exit(0);
}

int main()
{
        int device = open("/dev/input/event2", O_RDONLY);  // need a way to find keyboard at runtime
        struct input_event ev;
        signal(SIGINT, INThandler);
        while(1)
        {
                read(device,&ev, sizeof(ev));
               if(ev.type == EV_KEY && ev.value < 2)  // type is a KEY and value is 0 KeyUp or 1 KeyDown
		{
                        printf("RawCode: %4i Value: %4i \n", ev.code,ev.value);
                }
        }
}
```
This correctly reads my keyboard and all the constants are loaded from linux/input-event-codes.h I had a translator from Raw Code to strings but that is very long array definition.

This code would need to run in a thread, and somehow fill the  previousKeyState &  currentKeyState arrays if I'm understanding this aspect of the library correctly?

I don't know if this helps at all, but I really like this library and thought it might.

**Comment by Berni8k** [comment]
Should now be fixed with pull #785

**Comment by raysan5** [comment]
Verified! It works! :D
